### PR TITLE
✨amp-recaptcha-input: Added Reference Documentation.

### PR DESCRIPTION
**Pull request by torch2424**
Adds reference documentation to `amp-recaptcha-input`. Migrated from the Google Doc.

@CrystalFaith I reworked the first sentence after the description by your suggestion. Also, I added our current local development example in the example field, as I figured it would be better than having nothing until the ABE is launched. Please take a look at that 😄

### CODE REVIEW

File: extensions/amp-recaptcha-input/amp-recaptcha-input.md [hunk 1]
```diff
@@ -13,3 +13,82 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
+<td width="40%"><strong>Examples</strong></td>
+<td><a href="https://github.com/ampproject/amphtml/blob/master/examples/recaptcha.amp.html">Development Example</a></td>
+</tr>
+</table>
+
+[TOC]
+
+## Behavior
+
+This extension adds a parameter containing a recaptcha response token when a parent `<form>` element submits. `amp-recaptcha-input` does this by creating an iframe to load the reCAPTCHA v3 api script using the provided site key, and calling greaptcha.execute with the provided site key and action.
```
Comment (CrystalFaith) [inline extensions/amp-recaptcha-input/amp-recaptcha-input.md]: I think `greaptcha.execute` was intended to be `grecaptcha.execute`? And can you wrap it in ``` please :)

File: extensions/amp-recaptcha-input/amp-recaptcha-input.md [hunk 2]
```diff
@@ -13,3 +13,82 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
+
+### Example
```
Comment (CrystalFaith) [inline extensions/amp-recaptcha-input/amp-recaptcha-input.md]: Quick sample question - is there a specific place the `<amp-recaptcha-input>` should be placed in `<amp-form>` and does it have any purpose outside of forms?

Comment (torch2424) [reply extensions/amp-recaptcha-input/amp-recaptcha-input.md]: Good question! Updated the docs to try and make that more clear. Let me know what you think 😄

File: extensions/amp-recaptcha-input/amp-recaptcha-input.md (hunk 1 above)
Comment (torch2424) [reply extensions/amp-recaptcha-input/amp-recaptcha-input.md]: Done 😄 I went ahead and wrapped all `grecaptcha` in backticks as well 😄

**Comment by torch2424** [comment]
@CrystalFaith 

No worries! I was thinking of doing the `...` because I do that for all of my personal side projects, but I thought I was the only one 😂 Glad that you added it!

Thanks for the review!

Merging...

**Pull request merged**